pkg/gui/presentation/commits.go

```diff
@@ -99,6 +101,24 @@ func GetCommitListDisplayStrings(
 		getGraphLine = func(idx int) string { return "" }
 	}
 
+	// Determine the hashes of the local branches for which we want to show a
+	// branch marker in the commits list. We only want to do this for branches
+	// that are not the current branch, and not any of the main branches. The
+	// goal is to visualize stacks of local branches, so anything that doesn't
+	// contribute to a branch stack shouldn't show a marker.
+	branchHeadsToVisualize := lo.SliceToMap(lo.FilterMap(branches,
```

Comment:
you can also use the Set type from my generics package via `set.NewFromSlice(...)` and then later on go:
```go
} else if branchHeadsToVisualize.Includes(commit.Sha) &&
```
Up to you